### ISSUE TITLE
λ G L G, the loss2(maximizing the inter-class distances) could not decline?

### ISSUE THREAD

**Issue by rain2008204** [kjanjua26/Git-Loss-For-Deep-Face-Recognition#2]
The loss2 always floating in a small range? how can solve it?

![default](https://user-images.githubusercontent.com/34569278/44889795-206b9600-ad0a-11e8-9bd7-ef4786b0e3b3.png)

@Shahnawazgrewal @kjanjua26

**Comment by kjanjua26** [comment]
@rain2008204 Are you just training gitloss + softmax, because loss2 is gitloss loss defined in `get_git_loss()` function. Or is it centerloss + gitloss + softmax?

**Comment by rain2008204** [comment]
@kjanjua26  return combo_loss, centers_update_op， I just modify it.
return combo_loss, centers, FLAGS.lambda_c * loss1, FLAGS.lambda_g * loss2.
why the loss2 can't decline, just floating?

**Comment by kjanjua26** [comment]
@rain2008204 the thing is gitloss is a combination of loss1 and loss2. With the combination minimization of intra-class variations and maximization of inter-class distances is obtained. Loss1 is for one aspect and loss2 for other. So, it is important to train the network in combination fashion, as the paper explains. Similarly, if you remove softmax and train loss1+loss2, you wouldn't get classification results, for obvious reasons.

**Comment by kjanjua26** [comment]
@rain2008204 if your question has been answered, please close the issue. If not, reach out for further information. 
Thanks.
